### PR TITLE
Bug fix user update ft signup #175743313

### PR DESCRIPTION
**Pull request by kabano89**
#### What does this PR do?
- Fix user update details (**firstname, lastname, password**).

#### Description of Task to be completed? 
- Allows user to update his details
- A user cannot update his email
- Email updates are refused inside the database

#### How should this be manually tested?
- Clone the repository.

- In your home directory move into the cloned folder by cd this branch **`bug-fix-user-update-ft-signup-#175743313`**.

- Then run **`npm install`** to install all dependencies.

- Run **`npm migrate`** to create tables.

- Run **`npm start`** to start the server.

#### Screenshots (if appropriate)

- N/A

#### Questions:

- N/A

### CODE REVIEW

**Review by ConstantineMaliza — APPROVED** [review]

File: src/controllers/user.js [hunk 1]
```diff
@@ -102,16 +102,22 @@ export const getUserById = async (req, res) => {
 export const updateUserById = async (req, res) => {
     try {
       const id = req.params.id;
-      const [user] = await model.User.update(req.body, {where: {id : id }});
+      const [user] = await model.User.update(
+        {
+          firstname: req.body.firstname,
+          lastname: req.body.lastname,
+          password: req.body.password
```
Comment (SimonImanigirimpuhwe) [inline src/controllers/user.js]: This current implementation is causing a terrible bug coz can change password and store it as plain text
Screen Shot 2021-02-19 at 18 34 25

**Review by SimonImanigirimpuhwe — CHANGES_REQUESTED** [review]
Please rework on this fix and also consider the issue that any user is getting all users

**Review by furebo — CHANGES_REQUESTED** [review]
Please you can rebase and solve conflicts. Then remember to encrypt the password when you are creating and updating a user before it is saved to database.
Also where you are using jwt.verify(...) there is a third parameter that is a call back function which is doing nothing , I think you can remove it. Thank you and good work.

File: src/controllers/user.js (hunk 1 above)
Comment (kabano89) [reply src/controllers/user.js]: Thx @SimonImanigirimpuhwe I have added the encryption before we update the password

**Review by furebo — APPROVED** [review]
Good work now the password is hashed before it is saved

**Comment by kabano89** [comment]
> Good work now the password is hashed before it is saved

Thx

**Review by thepatrickniyo — APPROVED** [review]

File: locales/en.json [hunk 2]
```diff
@@ -73,6 +73,12 @@
 	"Password was reset successfully": "Password was reset successfully",
 	"Check your email for the reset link": "Check your email for the reset link",
 	"You attempt to assign a room to the hotel which does not exist ! Room not created.": "You attempt to assign a room to the hotel which does not exist ! Room not created.",
+	"User updated successfully": "User updated successfully",
```
Comment (SamuelSabasNayo) [inline locales/en.json:76]: Please correctly here, there is a missing message.

**Review by SamuelSabasNayo — APPROVED** [review]

**Review by SimonImanigirimpuhwe — APPROVED** [review]

**Review by SamuelSabasNayo — COMMENTED** [review]
Good work!

**Pull request merged**